src/adapters/github.ts

```diff
@@ -18,6 +19,7 @@ enum SafetyStatus {
 
 class GithubAdapter extends GitAdapter {
   private octokit: Octokit;
+  private static validSearchTypes = new Set(['code', 'repositories']);
```

Comment:
Micro-nit: Might be more conventional to declare this outside the scope of the class instead of declaring it as a static member, i.e.

```ts
const VALID_SEARCH_TYPES = new Set(['code', 'repositories']);

class GithubAdapter extends GitAdapter {
  // ...
}
```

Should make it slightly clearer that this is a constant. Better yet, we can include a const assertion (after converting to an array instead of a set:

```ts
const VALID_SEARCH_TYPES = ['code', 'repositories'] as const;
```

Reply:
I prefer the suggested way as well, just thought I'd try to be `class-y` :smile:.

Reply:
This is complete, resolving.